### ISSUE TITLE
Sourcing inventory from EC2 hangs with 'overwrite' due to host being in use 

### ISSUE THREAD

**Issue by endriu0** [ansible/awx#9555]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
AWX Inventory Sourcing from EC2 hangs on refresh because a node appears to be in use by another playbook and cannot be deleted when "Overwrite" is chosen.

##### ENVIRONMENT
* AWX version: 17.0.1
* AWX install method: k8s
* Ansible version:  2.9.17
* Operating System: MacOS
* Web Browser: Chrome

##### STEPS TO REPRODUCE

1. Cause AWX to think a playbook is blocking host from being deleted.
2. Try and run a sync of AWS EC2 with "overwrite" specified
3. Sync hangs forever - on info log level not giving any info except just new nodes being imported / on debug showing a delete of some hosts going through and then just stopping.

##### EXPECTED RESULTS

Inventory source sync to go through and current list of hosts to be replaced by what's in AWS atm.

##### ACTUAL RESULTS

Because a task was run as a callback from host this task then waited forever for the inventory source sync to finish blocking all of the next runs of any new playbooks and compounding the issue.

##### ADDITIONAL INFORMATION

After cancelling the sync from inventory source it was still hanging. Cancelling it from list of current jobs also did nothing. Only solution to actually remove it and unblock new instance refreshes was using manage_shell to delete the specific id as per this comment : https://github.com/ansible/awx/issues/888#issuecomment-358639967

**Comment by shanemcd** [comment]
Hi @endriu0 - can you check and see if this happens on AWX $latest?

**Comment by endriu0** [comment]
Hey @shanemcd . I can confirm it's changed - no longer hanging forever instead throwing a 502 after a minute or so on the `broken` hosts. Found it to be only the case with AWS EC2 sourced inventories and cannot really figure out what's causing some hosts to be in the broken state. 

From my testing trying multiple times between gui / api finally removes the broken hosts and it just works on next pull. Could be something that broke with ASG scale up. Will keep an eye and open a new issue if I can get more info,.
